homeassistant/components/yardian/__init__.py

```diff
@@ -0,0 +1,87 @@
+"""The Yardian integration."""
+from __future__ import annotations
+
+from pyyardian import AsyncYardianClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN, MAX_ZONES
+from .coordinator import YardianUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional("device", default=[]): vol.Schema(
+                    [
+                        {
+                            vol.Required("yid"): cv.string,
+                            vol.Required("host"): cv.string,
+                            vol.Required("access_token"): cv.string,
+                            vol.Optional("duration"): vol.Schema(
+                                {
+                                    vol.Optional(i + 1): cv.positive_int
+                                    for i in range(MAX_ZONES)
+                                },
+                                extra=vol.ALLOW_EXTRA,
+                            ),
+                        }
+                    ]
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up for Yardian."""
+
+    hass.data[DOMAIN] = {
+        "config": config.get(DOMAIN, {}),
+    }
+
+    return True
```

Comment:
```suggestion
```

We don't use YAML config for new device integrations

Reply:
Got it. 
Does that mean I should find another way to let user having some customized settings or it's just I should't use YAML config in the first PR?

Thanks 😃

Reply:
After this PR is merged you can add an options flow

https://developers.home-assistant.io/docs/config_entries_options_flow_handler?_highlight=options&_highlight=flow